tools/diagnostics/agent.py

```diff
@@ -0,0 +1,150 @@
+import logging
+import os
+import re
+import retrying
+from typing import List
+
+import sdk_cmd
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_RETRY_WAIT = 1000
+DEFAULT_RETRY_MAX_ATTEMPTS = 5
+
+
+def is_http_server_error(http_status_code: int) -> bool:
+    return http_status_code >= 500 and http_status_code <= 599
+
+
+@retrying.retry(wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS)
+def debug_agent_files(agent_id: str) -> List[str]:
+    response = sdk_cmd.cluster_request(
+        "GET",
+        "/slave/{}/files/debug".format(agent_id),
+        retry=False,
+        raise_on_error=False,
+        log_response=False,
+    )
+
```

Comment:
Can you drop these empty lines?

Reply:
I find that vertical whitespace sometimes helps with readability, this case being an example. Since this was formatted with `black` does it make sense that we attempt to make it always remove all vertical whitespace inside functions? By default it preserves whatever it's there.

Reply:
I also agree whitespace sometimes helps with readability, but mostly in longer functions, so I wouldn't necessarily make black remove it unconditionally...
I just didn't feel like it was helping in this particular case. But if you feel stronger about this then feel free to leave them in.

Reply:
Ok.